src/query/api/v1/handler/search.go

```diff
@@ -90,24 +98,19 @@ func (h *SearchHandler) parseBody(r *http.Request) (*storage.FetchQuery, *xhttp.
 	return &fetchQuery, nil
 }
 
-func (h *SearchHandler) parseURLParams(r *http.Request) *storage.FetchOptions {
-	var (
-		limit int
-		err   error
-	)
-
-	limitRaw := r.URL.Query().Get("limit")
-	if limitRaw != "" {
-		limit, err = strconv.Atoi(limitRaw)
-		if err != nil {
-			limit = defaultLimit
+func (h *SearchHandler) parseURLParams(r *http.Request) (*storage.FetchOptions, *xhttp.ParseError) {
+	fetchOpts, err := h.fetchOptionsBuilder.NewFetchOptions(r)
+	if err != nil {
+		return nil, err
+	}
+
+	if str := r.URL.Query().Get("limit"); str != "" {
+		if limit, err := strconv.Atoi(str); err == nil {
```

Comment:
honestly if they pass the correct query URL name but we can't parse it into a number, it should just return an error

Reply:
Yeah agreed, will change what this behavior used to do to match that.